simulator/src/node.js

```
@@ -71,6 +71,24 @@ export function findNode(x) {
 }
 
+/**
+ * Propagates updated bitWidth across connected intermediate wire nodes.
+ * @param {Node} startNode
+ * @param {number} bitWidth
+ * @param {Set} visited
+ */
+export function propagateBitWidth(startNode, bitWidth, visited = new Set()) {
+    if (!startNode || visited.has(startNode)) return;
+    visited.add(startNode);
+    startNode.bitWidth = bitWidth;
```

ESLint

[error] Assignment to property of function parameter 'startNode'.

(no-param-reassign)